spec/operators/splitBy-spec.ts

```diff
@@ -0,0 +1,35 @@
+/** @prettier */
+import { merge, of } from 'rxjs';
+import { map, mergeMap, splitBy } from 'rxjs/operators';
+import { TestScheduler } from 'rxjs/testing';
+import { observableMatcher } from '../helpers/observableMatcher';
+
+describe('splitBy()', () => {
+  let rxTest: TestScheduler;
+
+  beforeEach(() => {
+    rxTest = new TestScheduler(observableMatcher);
+  });
+
+  it('should split values', () => {
+    rxTest.run(({ cold, expectObservable }) => {
+      const source = cold(' --a-b-c-a-b-c-|');
+      const x = cold('      --a-----a-----|');
+      const y = cold('      ----b-c---b-c-|');
+      const expected = '    (xy)----------|';
+
+      const split = source.pipe(
+        splitBy((value) => value === 'a'),
+        mergeMap((splits) => splits)
+      );
+      expectObservable(split).toBe(expected, { x, y });
+    });
+  });
+
+  it('should infer partition types when using a type guard predicate', () => {
+    of(5, 'a', []).pipe(
+      splitBy((value): value is number => typeof value === 'number'),
+      mergeMap(([numbers, strings]) => merge(numbers.pipe(map((value) => value * value)), strings.pipe(map((value) => value.length))))
+    );
+  });
```

Comment:
We favour using dtslint for type-related tests:

https://github.com/ReactiveX/rxjs/blob/c23e567ccaee4a9aa2a7046519e7cf46a7191a3f/spec-dtslint/operators/filter-spec.ts#L37-L40

The tests are never executed, instead there is a lint rule that runs evaluates the types and dtslint expectations. To run them, use `npm run dtslint`.